### PR TITLE
feat: add articles page

### PR DESCRIPTION
**Pull request by shahin-hq**
## Summary

- https://app.clickup.com/t/8678v0jj8
- https://app.clickup.com/t/8678va0kq
- https://app.clickup.com/t/8678va2qh
- https://app.clickup.com/t/8678v0jhh
- https://app.clickup.com/t/8678v0jgg
- https://app.clickup.com/t/8678va2rr
- https://app.clickup.com/t/8678v0jjz

## Checklist

-   [x] I checked my UI changes against the design and there are no notable differences
-   [x] I checked my UI changes for any responsiveness issues
-   [x] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
-   [ ] I added a short description on how to test this PR _(if necessary)_
-   [ ] I added a storybook entry for the component that was added _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [x] Tests _(if necessary)_
-   [x] Ready to be merged

### CODE REVIEW

File: resources/css/app.css [hunk 1]
```diff
@@ -39,3 +39,9 @@
         0 207px 83px 0 rgba(33, 42, 131, 0.01),
         0 323px 90px 0 rgba(33, 42, 131, 0);
 }
+
+.latest-articles-carousel .swiper-pagination-bullets {
+    display: flex;
+    justify-content: center;
+    bottom: 0;
```
Comment (ItsANameToo) [inline resources/css/app.css:46]: use tailwind classes

File: app/Http/Controllers/ArticleController.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    {
+        $pageLimit = min($request->has('pageLimit') ? (int) $request->get('pageLimit') : 10, 96);
+
+        //        Article::query()
+        //            ->search($request->get('search'))
+        //            ->when($request->get('sort') !== 'popularity', fn ($q) => $q->sortById())
+        //            ->with(['collections' => function ($query) {
+        //                $query->select(['collections.name', 'collections.extra_attributes->image as image']);
+        //            }])
+        //            ->dd();
```
Comment (ItsANameToo) [inline app/Http/Controllers/ArticleController.php:28]: ?

File: resources/js/Pages/Articles/Components/LatestArticles.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                    "-mx-6 -mt-6 space-y-2 border-b-4 border-theme-secondary-100 sm:-mx-8 lg:mx-0 lg:border-b-2 lg:border-theme-secondary-300":
+                        withFullBorder && (hasEnoughArticles || isLoading),
+                    "border-b-2 border-theme-secondary-300": !withFullBorder && (hasEnoughArticles || isLoading),
+                })}
+            ></div>
+        </div>
+    );
+};
+
+const useArticlesCarousel = (): number => {
```
Comment (ItsANameToo) [inline resources/js/Pages/Articles/Components/LatestArticles.tsx:70]: this name is odd as it returns a number of slides to use 🤔 based on the name I had expected something else like a boolean that determines whether it uses a carousel

File: app/Data/Articles/ArticleData.php [hunk 4]
```diff
@@ -47,7 +47,7 @@ public static function fromModel(Article $article): self
             category: $article->category,
             content: $article->content,
             image: $article->getMedia()->first()->getUrl(),
-            publishedAt: (int) $article->published_at->timestamp,
+            publishedAt: (int) $article->published_at->getTimestampMs(),
```
Comment (ItsANameToo) [inline app/Data/Articles/ArticleData.php:50]: why was this change needed?

**Review by ItsANameToo — CHANGES_REQUESTED** [review]
some general remarks already:
- when articles don't exist, their placeholders should not be animated as that makes it appear as if it's loading them. These should be static instead to keep a difference between loading states and placeholders
- search and filters should be disabled when no entries exist
image

- placeholder text should read something generic and not use the collection placeholder text as this page does not do anything with collections
- search has a brief moment where it shows the results immediately, then flashes a loading skeleton before showing the results again.
- can you adjust the seeder to make use of seeded collections so we can see them with images and proper links, as opposed to the factory ones it currently uses

File: resources/js/Pages/Articles/Components/LatestArticles.tsx (hunk 3 above)
Comment (shahin-hq) [reply resources/js/Pages/Articles/Components/LatestArticles.tsx]: Should be fine

File: resources/css/app.css (hunk 1 above)
Comment (shahin-hq) [reply resources/css/app.css]: Should be fine

File: app/Data/Articles/ArticleData.php (hunk 4 above)
Comment (shahin-hq) [reply app/Data/Articles/ArticleData.php:50]: because JS expects timestamp to be in milliseconds

Comment (ItsANameToo) [reply app/Data/Articles/ArticleData.php:50]: usually we do `* 1000` in the frontend to achieve that (e.g. `ResultStep.tsx` and various other places). Might be best to keep the timestamp as-is and handle it in the frontend to keep it consistent

Comment (shahin-hq) [reply app/Data/Articles/ArticleData.php]: Fixed. Thanks

**Review by ItsANameToo — APPROVED** [review]
